packages/backend/src/graphql/mutations/login-with-sgid.ts

```diff
@@ -0,0 +1,108 @@
+import { SgidClient } from '@opengovsg/sgid-client'
+
+import appConfig from '@/config/app'
+import { getOrCreateUser, setAuthCookie } from '@/helpers/auth'
+import { validateAndParseEmail } from '@/helpers/email-validator'
+import type Context from '@/types/express/context'
+
+const sgidClient = new SgidClient({
+  clientId: appConfig.sgid.clientId,
+  clientSecret: appConfig.sgid.clientSecret,
+  privateKey: appConfig.sgid.privateKey,
+  redirectUri: `${appConfig.webAppUrl}/login/sgid/redirect`,
+})
+
+interface PublicOfficerEmployment {
+  workEmail: string | null
+  agencyName: string | null
+  departmentName: string | null
+  employmentType: string | null
+  employmentTitle: string | null
```

Comment:
other than `workEmail` why do we need the rest of the employment data 🤔 ? maybe u could keep agencyName for easier disambiguation?

Reply:
This one is copied from SGC design - (in a later PR) we show the agency name, dept name and title when user chooses their email.

We can definitely leave out `employmentType`, but POCDEX returns this to us by default so I'm just leaving it in the type in case we ever need it <.<